src/Forms/GridField/GridFieldFilterHeader.php

```diff
@@ -15,10 +22,150 @@
  * GridFieldFilterHeader alters the {@link GridField} with some filtering
  * fields in the header of each column.
  *
+ * filter header with customisable filter fields
+ *
+ * REQUIREMENTS:
+ *
+ * table header component needs to be present in the GridField (for example GridFieldSortableHeader)
+ * the last column of the table needs to have a vacant header so the filter widget could be displayed there
+ * for example you can't have the last column with sorting header and filter widget at the same time
+ *
+ * ALIAS FIELDS
+ *
+ * in the case that the column name of the table is different from matching field name via the summary_fields
+ * a field alias needs to be specified
+ *
+ * configuration format:
+ *
+ * 'field_alias' => 'field_name'
+ *
+ * for example if summary fields looks like this:
+ *
+ * private static $summary_fields = [
+ *   'getTitleSummary' => 'Title',
+ *   'City.Name' => 'City',
+ *   'Expires.Nice' => 'Expires',
+ * ];
+ *
+ * alias fields need to be specified to match the summary fields mapping
+ *
+ * ->setAliasFields([
```

Comment:
If this matches the summary_fields exactly, why is it mandatory for the user to set the field aliases explicitly? Can't the component just infer from summary_fields directly?